### PR TITLE
Revert rspec as runtime dependency

### LINKED ISSUE #115: uninitialized constant RSpec::Matchers when running rake
Steps to reproduce

Gemfile:
```ruby
gem 'rails_event_store'

group :test do
   gem 'rspec-rails'
   gem 'rails_event_store-rspec'
end
```

```
RAILS_ENV=test rake db:create

rake aborted!
NameError: uninitialized constant RSpec::Matchers
/Users/michal/.rvm/gems/ruby-2.3.3/gems/rails_event_store-rspec-0.17.0/lib/rails_event_store/rspec/be_event.rb:115:in `<class:BeEvent>'
/Users/michal/.rvm/gems/ruby-2.3.3/gems/rails_event_store-rspec-0.17.0/lib/rails_event_store/rspec/be_event.rb:3:in `<module:RSpec>'
/Users/michal/.rvm/gems/ruby-2.3.3/gems/rails_event_store-rspec-0.17.0/lib/rails_event_store/rspec/be_event.rb:2:in `<module:RailsEventStore>'
/Users/michal/.rvm/gems/ruby-2.3.3/gems/rails_event_store-rspec-0.17.0/lib/rails_event_store/rspec/be_event.rb:1:in `<top (required)>'
/Users/michal/.rvm/gems/ruby-2.3.3/gems/rails_event_store-rspec-0.17.0/lib/rails_event_store/rspec.rb:8:in `<top (required)>'
```

or have a look at the sample app to reproduce the issue https://github.com/mlomnicki/res-rspec-error

The error goes away if I add 'rspec' to the Gemfile (rspec-rails is not enough)

```ruby
group :test do
   gem 'rspec'
   gem 'rspec-rails'
   gem 'rails_event_store-rspec'
end
```

### PR DESCRIPTION
**Pull request by jandudulski**
Without rspec gem tests are failing with LoadError:

    cannot load such file -- rspec

Regression fix #115

---
This change is [Reviewable](https://reviewable.io/reviews/railseventstore/rails_event_store/203)

### CODE REVIEW

**Comment by paneq** [comment]
https://stackoverflow.com/questions/24333152/difference-between-add-dependency-and-add-runtime-dependency

> TLDR: The same.
